test/OpenTelemetry.Exporter.Geneva.Tests/GenevaLogExporterAFDCorrelationTests.cs

```diff
@@ -0,0 +1,392 @@
+                        Assert.Equal(actualCorrelationId, expectedCorrelationId);
+                    }
+                    else
+                    {
+#pragma warning disable CA2254 // Template should be a static expression
+                        logger.LogInformation($"Thread {threadIndex} without correlation ID");
+#pragma warning restore CA2254 // Template should be a static expression
+                        lock (syncObj)
+                        {
+                            countWithoutCorrelationId++;
+                        }
+
+                        byte[] serializedData;
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        {
+                            serializedData = MsgPackLogExporter.Buffer.Value;
+                        }
+                        else
+                        {
+                            // Read the data sent via socket
+                            serializedData = new byte[65360];
+                            _ = receiverSocket.Receive(serializedData);
+                        }
+
+                        var fluentdData = MessagePack.MessagePackSerializer.Deserialize<object>(serializedData, MessagePack.Resolvers.ContractlessStandardResolver.Options);
+                        var signal = (fluentdData as object[])[0] as string;
+                        var TimeStampAndMappings = ((fluentdData as object[])[1] as object[])[0];
+                        var mapping = (TimeStampAndMappings as object[])[1] as Dictionary<object, object>;
+
+                        // Verify that AFDCorrelationId is not present in the serialized data
+                        Assert.False(mapping.ContainsKey("AFDCorrelationId"));
+                    }
+                });
+            }
+
+            // Start all threads
+            foreach (var thread in threads)
+            {
+                thread.Start();
+            }
+
+            // Wait for all threads to finish
+            foreach (var thread in threads)
+            {
+                thread.Join();
+            }
+
+            // Verify the correct number of logs were created
+            Assert.Equal(threadCount, exportedItems.Count);
+            Assert.Equal(threadCount / 2, countWithCorrelationId);
+            Assert.Equal(threadCount / 2, countWithoutCorrelationId);
+
+            // Check that no exceptions were thrown
+            // If our implementation is correct, logs from threads without correlation ID
+            // should have been processed without exceptions
+        }
+        finally
+        {
+            senderSocket?.Dispose();
+            receiverSocket?.Dispose();
+            try
+            {
+                File.Delete(path);
+            }
+            catch
+            {
+            }
+        }
+    }
+
+    [Fact]
+    public void AFDCorrelationIdLogProcessor_WithoutCorrelationId_HandlesGracefully()
```

Comment:
nit: the WithoutCorrelationId and WithCorrelationId tests are very similar. 
These could probably be combined into a `[Theory]` with a bool parameter to simplify this test file.

not blocking, could be done in a follow up PR.